test/sqlsmith/mzcompose.py

```diff
@@ -0,0 +1,175 @@
+    "nested aggregate functions are not allowed",
+    "is defined for numbers greater than or equal to",
+    "is not defined for zero",
+    "is not defined for negative numbers",
+    "requested character too large for encoding",
+    "internal error: unrecognized configuration parameter",
+    "invalid encoding name",
+    "invalid time zone",
+    "value out of range: overflow",
+    "LIKE pattern exceeds maximum length",
+    "negative substring length not allowed",
+    "cannot take square root of a negative number",
+    "timestamp units not yet supported",
+    "step size cannot equal zero",
+    "stride must be greater than zero",
+    "timestamp out of range",
+    "unterminated escape sequence in LIKE",
+    "null character not permitted",
+    "is defined for numbers between",
+    "field position must be greater than zero",
+    "' not recognized",  # Expected, see https://github.com/MaterializeInc/materialize/issues/17981
+    "must appear in the GROUP BY clause or be used in an aggregate function",
+    "Expected joined table, found",  # Should fix for multi table join
+    "Expected ON, or USING after JOIN, found",  # Should fix for multi table join
+    "but expression is of type",  # Should fix, but only happens rarely
+    "coalesce could not convert type map",  # Should fix, but only happens rarely
+    "operator does not exist: map",  # Should fix, but only happens rarely
+    "result exceeds max size of",  # Seems expected with huge queries
+    "expected expression, but found reserved keyword",  # Should fix, but only happens rarely with subqueries
+    "Expected right parenthesis, found left parenthesis",  # Should fix, but only happens rarely with cast+coalesce
+    "invalid selection: operation may only refer to user-defined tables",  # Seems expected
+]
+
+
+def is_known_error(e: str) -> bool:
+    for known_error in known_errors:
+        if known_error in e:
+            return True
+    return False
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    # parser.add_argument("--max-queries", nargs="?", default=None, type=int)
+    parser.add_argument("--runtime", nargs="?", default=None, type=int)
+    parser.add_argument("--max-joins", nargs="?", default=None, type=int)
+    parser.add_argument("--seed", nargs="?", default=None, type=int)
+    args = parser.parse_args()
+
+    c.up("materialized")
+
+    # Very simple data for our workload
+    c.sql(
+        """
+        CREATE SOURCE tpch
+          FROM LOAD GENERATOR TPCH (SCALE FACTOR 0.00001)
+          FOR ALL TABLES
+          WITH (SIZE = '1');
+
+        CREATE SOURCE auction
+          FROM LOAD GENERATOR AUCTION (SCALE FACTOR 0.01)
+          FOR ALL TABLES
+          WITH (SIZE = '1');
+
+        CREATE SOURCE counter
+          FROM LOAD GENERATOR COUNTER (SCALE FACTOR 0.0001)
+          WITH (SIZE = '1');"""
+    )
+
+    # queries = args.max_queries or 100000
+    runtime = args.runtime or 600
```

Comment:
You may wish to move those defaults up to the `add_argument` section. This way they will be printed on `--help`.